Library/Homebrew/test/caveats_spec.rb

```diff
@@ -26,4 +26,40 @@ def caveats
       expect(described_class.new(f)).not_to be_empty
     end
   end
+
+  describe "#caveats" do
+    context "when f.plist is not null" do
+      it "prints plist startup information when plist_startup is not null" do
+        allow(f).to receive(:plist).and_return(true)
+        allow(f).to receive(:plist_startup).and_return(true)
+        expect(subject.caveats).to include("To have launchd start #{f.full_name} now and restart at startup:\n  sudo brew services start #{f.full_name}\n")
+      end
+
+      it "prints plist login information when f.plist_startup is null" do
+        allow(f).to receive(:plist).and_return(true)
+        expect(subject.caveats).to include("To have launchd start #{f.full_name} now and restart at login:\n  brew services start #{f.full_name}\n")
+      end
+    end
+
+    context "when f.keg_only is not null" do
+      it "prints formula is keg_only when keg_only is not null" do
```

Comment:
Minor nitpick: we don't need `when keg_only is not null` since that's already a part of the `context` here.